src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/IApplyChangesToWorkspaceContext.cs

```diff
@@ -83,5 +83,14 @@ internal interface IApplyChangesToWorkspaceContext
         ///     intention that the <see cref="IWorkspaceProjectContext"/> will be immediately disposed.
         /// </remarks>
         Task ApplyProjectBuildAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> update, ContextState state, CancellationToken cancellationToken);
+
+        /// <summary>
+        ///     Sets project build command line options to the underlying <see cref="IWorkspaceProjectContext"/>.
+        /// </summary>
+        /// <param name="arguments">List of command line arguments in the Build</param>
+        /// <remarks>
+        ///     Note: The compiler is sensitive to ordering of command-line arguments.
+        /// </remarks>
+        void SetCommandLineArgumentsToBuild(IEnumerable<string> arguments);
```

Comment:
Rather than adding a new member here, it'd be better to pass the project build snapshot as part of `ApplyProjectBuildAsync`. If you look at `ApplyProjectChangesUnderLockAsync` you will see it is called within a lock, and does a bunch of other scheduling (eg. start/end batch). Making another method here and calling it out of sync with `ApplyProjectBuildAsync` will lead to data races and potentially other kinds of issues.

Reply:
Done, please latest commit "Sync-link buildSnapshot with the ProjectBuildRuleSource."